diff -u wcwidth-cjk-master.orig/Makefile wcwidth-cjk-master/Makefile
--- wcwidth-cjk-master.orig/Makefile	2013-08-08 23:53:35.000000000 +0900
+++ wcwidth-cjk-master/Makefile	2014-03-12 23:34:50.000000000 +0900
@@ -1,8 +1,8 @@
-BUILD_TARGETS = wcwidth-cjk wcwidth-cjk.so
+BUILD_TARGETS = wcwidth-cjk libwcwidth-cjk.dylib
 
 CC = gcc
 CFLAGS = -O2 -Wall -fPIC -Dwcwidth_cjk=wcwidth -Dwcswidth_cjk=wcswidth
-LDSHARED = $(CC) -shared
+LDSHARED = $(CC) -dynamiclib
 LDFLAGS =
 
 prefix = /usr/local
@@ -14,15 +14,15 @@
 
 clean:
 	rm -f $(BUILD_TARGETS)
-	rm -f *.o *.so *.tmp
+	rm -f *.o *.dylib *.tmp
 
 install:
 	mkdir -m 0755 -p $(DESTDIR)$(bindir)
 	rm -f $(DESTDIR)$(bindir)/wcwidth-cjk
 	cp wcwidth-cjk $(DESTDIR)$(bindir)/
 	mkdir -m 0755 -p $(DESTDIR)$(libdir)
-	rm -f $(DESTDIR)$(libdir)/wcwidth-cjk.so
-	cp wcwidth-cjk.so $(DESTDIR)$(libdir)/
+	rm -f $(DESTDIR)$(libdir)/libwcwidth-cjk.dylib
+	cp libwcwidth-cjk.dylib $(DESTDIR)$(libdir)/
 
 build: $(BUILD_TARGETS)
 
@@ -31,6 +31,6 @@
 	chmod +x $@.tmp
 	mv $@.tmp $@
 
-wcwidth-cjk.so: wcwidth.o
+libwcwidth-cjk.dylib: wcwidth.o
 	$(LDSHARED) $(LDFLAGS) -o $@ wcwidth.o
 
diff -u wcwidth-cjk-master.orig/wcwidth-cjk.sh wcwidth-cjk-master/wcwidth-cjk.sh
--- wcwidth-cjk-master.orig/wcwidth-cjk.sh	2013-08-08 23:53:35.000000000 +0900
+++ wcwidth-cjk-master/wcwidth-cjk.sh	2014-03-12 23:29:32.000000000 +0900
@@ -5,23 +5,25 @@
   exit 1
 fi
 
-wcwidth_cjk_so="@libdir@/wcwidth-cjk.so"
+wcwidth_cjk_so="@libdir@/libwcwidth-cjk.dylib"
 
-case ":$LD_PRELOAD:" in
+case ":$DYLD_INSERT_LIBRARIES:" in
 *":$wcwidth_cjk_so:"*)
   ;;
 *)
-  LD_PRELOAD="${LD_PRELOAD:+$LD_PRELOAD:}$wcwidth_cjk_so"
+  DYLD_INSERT_LIBRARIES="${DYLD_INSERT_LIBRARIES:+$DYLD_INSERT_LIBRARIES:}$wcwidth_cjk_so"
   ;;
 esac
 
 if [ x"$1" = x"--sh-init" ]; then
-  echo "LD_PRELOAD='$LD_PRELOAD'"
-  echo "export LD_PRELOAD"
+  echo "DYLD_INSERT_LIBRARIES='$DYLD_INSERT_LIBRARIES'"
+  echo "export DYLD_INSERT_LIBRARIES"
   exit 0
 fi
 
-export LD_PRELOAD
+DYLD_FORCE_FLAT_NAMESPACE=1
+export DYLD_FORCE_FLAT_NAMESPACE
+export DYLD_INSERT_LIBRARIES
 
 exec "$@"
 exit 1
